### PR TITLE
Create CODEOWNERS

### LINKED ISSUE #1633: "Feature maintainers" groups to collaboratively manage IS
Similar to the one discussed in  publiclab/plots2#6501 we should have maintainers groups in IS to collaboratively manage different features of IS this would help in maintaining the library more efficiently, Features around which we can develop a team around includes:

- [ ] `core-maintainers`:
**GOALS:** Stability and optimization of the core library.
**CURRENT PRIORITIES**: Fix deprecated libraries. eg: readPixels and savePixels are actually not maintained and in fact, do not work at all for GIFs.

- [ ] `module-maintainers`:
**GOALS**: Increasing the number of modules, stability/optimization of existing modules. Ensuring that creating new modules remains easy and accessible and well documented. Developing and maintaining compelling use-cases for IS, as in #1480.
**CURRENT PRIORITIES**: Increasing module GIF support and fixing bugs.

- [ ] `ui-maintainers`
**GOALS**: Stability and performance.
**CURRENT PRIORITIES**: Fixing UI bugs and perhaps refactoring some of the old, unmanageable code.

- [ ] `tests-maintainers`:
**GOALS**: Increasing test coverage.
**CURRENT PRIORITIES**: Increasing UI test coverage. Module and GIF tests are messy and decoupled, this needs to change as these tests are breaking unexpectedly.

- [ ] `cli-maintainers`:
**GOALS**: maintain existing features and expand test suite.
**CURRENT PRIORITIES**: completing initial CLI test suite and covering all basic usages

What do you guys think?

@publiclab/is-reviewers @publiclab/support @publiclab/mentors

cc: @jywarren

### PR DESCRIPTION
**Pull request by HarshKhandeparkar**
Fixes #1633 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm run test-all`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] ask `@publiclab/is-reviewers` for help, in a comment below
* [ ] Insert-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software
Please make sure to get at least two reviews before asking for merging the PR as that would make the PR more reliable on our part
Thanks!

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+/src/*.json @publiclab/is-core-maintainers
+/src/util/ @publiclab/is-core-maintainers
+
+/src/modules/ @publiclab/is-module-maintainers
+/docs/ @publiclab/is-module-maintainers
+
+/test/ @publiclab/is-tests-maintainers
+/jest* @publiclab/is-tests-maintainers
+/eslint* @publiclab/is-tests-maintainers
+/travis* @publiclab/is-tests-maintainers
```
Comment (jywarren) [inline .github/CODEOWNERS:64]:
```suggestion
/travis* @publiclab/is-tests-maintainers
.travis.yml
.gitpod*
```

Comment (jywarren) [reply .github/CODEOWNERS:64]: what do you think?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+/gitpod* @publiclab/is-tests-maintainers
+
+/examples/ @publiclab/is-ui-maintainers
+/icons/ @publiclab/is-ui-maintainers
```
Comment (jywarren) [inline .github/CODEOWNERS:68]:
```suggestion
/icons/ @publiclab/is-ui-maintainers
/test/ui-2/test/*
/test/ui/spec/*
```

adding here so ui maintainers see changes to ui tests!

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+# <-- /COMMON TO ALL MAINTAINERS -->
+
+# <-- SPECIFIC MAINTAINERS -->
+/index.js @publiclab/is-cli-maintainers
+/src/cli/ @publiclab/is-cli-maintainers
+/src/CliUtils.js @publiclab/is-cli-maintainers
+
+/src/*.js @publiclab/is-core-maintainers
+/src/*.json @publiclab/is-core-maintainers
+/src/util/ @publiclab/is-core-maintainers
```
Comment (jywarren) [inline .github/CODEOWNERS:56]:
```suggestion
/src/util/ @publiclab/is-core-maintainers
/test/core/* @publiclab/is-core-maintainers
```

Comment (jywarren) [reply .github/CODEOWNERS:56]: so, having test maintainers makes it a bit weird - like, they would take precedence and core maintainers wouldn't get notified here, i think?

**Review by jywarren — COMMENTED** [review]
OK, added some ideas, but the precedence thing is a little weird. What do you think?

if a PR modifies both a test/core directory and a core file, would both get notified?

File: .github/CODEOWNERS (hunk 1 above)
Comment (HarshKhandeparkar) [reply .github/CODEOWNERS:64]: Oh, I guess I made a mistake. I wanted `.travis*` or something like that... Yeah, this looks good, I'll commit.

File: .github/CODEOWNERS (hunk 3 above)
Comment (HarshKhandeparkar) [reply .github/CODEOWNERS:56]: Ok

**Comment by HarshKhandeparkar** [comment]
I committed your changes @jywarren.

**Comment by jywarren** [comment]
Ah, i'm seeing the electron download issue that's also in #1718 -- cc @daemon1024 that this is system-wide, its probably the link is no longer good for the .zip download:

> Error: GET https://github.com/electron/electron/releases/download/v5.0.13/electron-v5.0.13-linux-x64.zip returned 429

https://travis-ci.com/github/publiclab/image-sequencer/jobs/399672398#L668

we should try resolving this and it may unblock both PRs.

**Comment by jywarren** [comment]
https://github.com/electron/electron/releases/

**Comment by jywarren** [comment]
Could this be linked to https://github.com/publiclab/image-sequencer/pull/55#issuecomment-314518269 with `tape-run`?

We recently upgrade to tape-run v7, let's try to go to v8 which will rely on a more recent version of electron:

https://github.com/publiclab/image-sequencer/pull/1662

**Comment by HarshKhandeparkar** [comment]
Maybe a bump? puppeteer version bump might work?

**Comment by jywarren** [comment]
I just bumped dependabot, but don't see anything yet...

**Comment by HarshKhandeparkar** [comment]
Electron downloads are always weird...

**Comment by jywarren** [comment]
ok, tape-run v8 installed!

**Comment by jywarren** [comment]
OK, let's do it!

**Pull request merged**